### PR TITLE
Migrate refs of create_post component

### PR DESCRIPTION
**Pull request by sridhar02**
#### Summary

Migrates the refs of textbox in create_post.jsx to new syntax 

#### Ticket Link
 Fixes https://github.com/mattermost/mattermost-server/issues/15796

### CODE REVIEW

File: components/create_post/create_post.jsx [hunk 1]
```diff
@@ -327,6 +327,7 @@ class CreatePost extends React.PureComponent {
         this.lastChannelSwitchAt = 0;
         this.draftsForChannel = {};
         this.lastOrientation = null;
+        this.textbox = React.createRef();
```
Comment (Willyfrog) [inline components/create_post/create_post.jsx:330]: can you rename it for consistency with other PRs
```suggestion
        this.textboxRef = React.createRef();
```

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Thanks for contributing! it is looking good, but can we rename it for consistency?

**Review by nevyangelova — APPROVED** [review]
Thanks! LGTM apart from @Willyfrog 's comment

**Review by Willyfrog — CHANGES_REQUESTED** [review]
there are other string references in this file, would you mind to migrate those too so we don't have to go over this same file again?
I can see: fileUpload, topDiv and createPostControls

**Review by Willyfrog — COMMENTED** [review]
Looking good, could you please remove the eslint disabling comment to prevent people from adding string refs in the future. it looks like:
```/* eslint-disable react/no-string-refs */```

**Review by Willyfrog — APPROVED** [review]
LGTM! thanks for your contribution!

**Review by stevemudie — APPROVED** [review]
LGTM

**Pull request merged**